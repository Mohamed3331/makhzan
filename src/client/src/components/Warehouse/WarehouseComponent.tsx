import React, { useState } from 'react'
import { Typography, Paper, Button, Skeleton, Stack } from '@mui/material'
import WarehouseOutlinedIcon from '@mui/icons-material/WarehouseOutlined'
import SquareFootSharpIcon from '@mui/icons-material/SquareFootSharp'
import { Swiper, SwiperSlide } from 'swiper/react'
import { Box } from '@mui/system'
import Image1 from '../../Assets/image1.jpeg'
import Image2 from '../../Assets/image2.jpeg'
import Image3 from '../../Assets/image3.jpeg'

const SwiperImageHeight = '180'

const WarehouseComponent = () => {
  const [loading, setLoading] = useState(false)

  if (loading) {
    return (
      <Stack spacing={1}>
        <Box sx={{ display: 'flex', justifyContent: 'space-around', alignItems: 'center' }}>
          <Skeleton variant="circular" width={150} height={150} />
          <Skeleton variant="rectangular" width={600} height={140} />
        </Box>
      </Stack>
    )
  }

  return (
    <div>
      <Box sx={{ maxWidth: '90%' }}>
        <Paper variant="outlined" square sx={{ display: 'flex', flexDirection: 'row' }}>
          <Swiper
            className="swiper-override-style"
            spaceBetween={0}
            slidesPerView={1}
            // onSlideChange={() => console.log('slide change')}
            // onSwiper={(swiper) => console.log(swiper)}
            pagination={true}
            loop
            navigation
          >
            <SwiperSlide>
              <img height={SwiperImageHeight} src={Image1} alt="" />
            </SwiperSlide>
            <SwiperSlide>
              <img height={SwiperImageHeight} src={Image2} alt="" />
            </SwiperSlide>
            <SwiperSlide>
              <img height={SwiperImageHeight} src={Image3} alt="" />
            </SwiperSlide>
          </Swiper>
          <Box
            sx={{
              display: 'flex',
              flexDirection: 'column',
              textAlign: 'left',
              justifyContent: 'space-around',
              marginX: '1rem',
              width: '100%',
            }}
          >
            <Typography variant="h5" component="div">
              3,150,000 EGP / Month
            </Typography>
            <Typography variant="body2" color="text.secondary">
              fully finish apartment at mivida emaar low price emaar low price
            </Typography>
            <Box
              sx={{
                maxWidth: '50%',
                display: 'flex',
                justifyContent: 'space-between',
              }}
            >
              <Box sx={{ display: 'flex', alignItems: 'center' }}>
                <WarehouseOutlinedIcon />
                <Typography variant="body2" color="text.secondary">
                  Warehouse
                </Typography>
              </Box>
              <Box sx={{ display: 'flex', alignItems: 'center' }}>
                <SquareFootSharpIcon />
                <Typography variant="body2" color="text.secondary">
                  152 sqm
                </Typography>
              </Box>
            </Box>
            <Typography variant="body2" color="text.secondary">
              hadayek el ahraam, street 105, apartment 15
            </Typography>
            <Button sx={{ marginBottom: '0.5rem' }} variant="contained">
              More Details
            </Button>
          </Box>
        </Paper>
      </Box>
    </div>
  )
}

export default WarehouseComponent
