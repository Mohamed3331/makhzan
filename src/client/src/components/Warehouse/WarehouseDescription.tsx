import React, { useState } from 'react'
import {
  Box,
  Container,
  ImageListItem,
  Avatar,
  Grid,
  Typography,
  Divider,
  Paper,
  Button,
  Stack,
} from '@mui/material'
import CallIcon from '@mui/icons-material/Call'
import EmailIcon from '@mui/icons-material/Email'
import PowerIcon from '@mui/icons-material/Power'
import WcIcon from '@mui/icons-material/Wc'
import WifiIcon from '@mui/icons-material/Wifi'
import AcUnitIcon from '@mui/icons-material/AcUnit'
import CompassCalibrationIcon from '@mui/icons-material/CompassCalibration'
import AccountCircleIcon from '@mui/icons-material/AccountCircle'

import { filteredServices } from '../../utils/filterServices'

const DividerComponent: React.FC = (props) => (
  <Divider sx={{ marginY: '10px' }}>
    <Typography variant="h6" component="div">
      {props.children}
    </Typography>
  </Divider>
)

const WarehouseDescription: React.FC = () => {
  const myServices = filteredServices(['Wifi', 'Thermostat'])
  return (
    <>
      <Box sx={{ marginTop: '2rem', textAlign: 'left' }}>
        <Grid columnSpacing={{ xs: 1, sm: 2, md: 1 }} container>
          <Grid item xs={8}>
            <Typography variant="body2" color="text.secondary">
              {'APARTMENT FOR SALE IN MIVIDA, 5TH SETTLEMENT COMPOUNDS'.toUpperCase()}
            </Typography>
            <Typography variant="h5" component="div">
              fully finish apartment at mivida emaar low price
            </Typography>
            <DividerComponent>Services</DividerComponent>
            {myServices.map(({ label, Icon }, indx: number) => {
              return (
                <Box key={indx} sx={{ display: 'flex' }}>
                  {<Icon />}
                  <Typography
                    sx={{ marginLeft: '10px', marginBottom: '15px' }}
                    variant="body1"
                    component="div"
                  >
                    {label.toUpperCase()}
                  </Typography>
                </Box>
              )
            })}
            <DividerComponent>Agent Information</DividerComponent>
            <Box
              sx={{
                display: 'flex',
                flexDirection: 'column',
                justifyContent: 'center',
                alignItems: 'center',
              }}
            >
              <div>
                <Avatar sx={{ width: 56, height: 56 }}>
                  <AccountCircleIcon fontSize="large" />
                </Avatar>
                haidy ahmed
              </div>
            </Box>
            <DividerComponent>Description</DividerComponent>
            {/* whiteSpace: 'pre-line' */}
            apartment for sale in mivida emaar new cairo with very special finishing at lowest price
            in market compound : mivida one of the luxuries compounds in new cairo , it contain all
          </Grid>
          <Grid item xs={4}>
            <Paper
              variant="outlined"
              square
              sx={{
                display: 'flex',
                justifyContent: 'space-around',
                flexDirection: 'column',
                height: '13rem',
              }}
              elevation={2}
            >
              <Typography sx={{ textAlign: 'center' }} variant="h5" component="div">
                3,150,000 EGP / Month
              </Typography>
              <Stack direction="row" spacing={5} sx={{ justifyContent: 'center' }}>
                <Button variant="contained">
                  <CallIcon sx={{ marginRight: '10px' }} /> Call
                </Button>
                <Button variant="contained">
                  <EmailIcon sx={{ marginRight: '10px' }} /> Email
                </Button>
              </Stack>
            </Paper>
          </Grid>
        </Grid>
      </Box>
    </>
  )
}

export default WarehouseDescription
