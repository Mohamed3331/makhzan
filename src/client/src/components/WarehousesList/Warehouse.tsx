import React from 'react'
import { Typography, Paper, Button, Stack, useMediaQuery } from '@mui/material'
import WarehouseOutlinedIcon from '@mui/icons-material/WarehouseOutlined'
import SquareFootSharpIcon from '@mui/icons-material/SquareFootSharp'
import { Swiper, SwiperSlide } from 'swiper/react'

import { Box } from '@mui/system'
import { formatRentValue } from '../../utils/formatNumber'
import { Link } from 'react-router-dom'

import { WarehouseAttributes } from '../../types'

interface IWarehouseComponentProps extends WarehouseAttributes {}

const Warehouse: React.FC<IWarehouseComponentProps> = ({
  title,
  id,
  governorate,
  location,
  rent,
  size,
  images,
  street,
}) => {
  const matches = useMediaQuery('(min-width:700px)')

  const SwiperImageHeight = matches ? '180' : '450'

  return (
    <>
      <Paper
        variant="outlined"
        square
        sx={{ display: 'flex', flexDirection: matches ? 'row' : 'column' }}
      >
        <Swiper
          className="swiper-override-style"
          spaceBetween={0}
          slidesPerView={1}
          pagination={true}
          loop
        >
          {images.map((image: string, idx: number) => (
            <SwiperSlide key={idx}>
              <img height={SwiperImageHeight} src={image} alt="" />
            </SwiperSlide>
          ))}
        </Swiper>
        <Stack
          direction="column"
          spacing={1.5}
          sx={{ paddingX: '1rem', textAlign: 'left', width: '100%' }}
        >
          <Typography variant="h5" component="div">
            {formatRentValue(rent)} EGP / Month
          </Typography>
          <Typography variant="body2" color="text.secondary">
            {title}
          </Typography>
          <Box sx={{ display: 'flex' }}>
            <Box sx={{ display: 'flex', alignItems: 'center' }}>
              <WarehouseOutlinedIcon />
              <Typography variant="body2" color="text.secondary">
                Warehouse
              </Typography>
            </Box>
            <Box sx={{ display: 'flex', alignItems: 'center' }}>
              <SquareFootSharpIcon />
              <Typography variant="body2" color="text.secondary">
                {size} sqm
              </Typography>
            </Box>
          </Box>
          <Typography variant="body2" color="text.secondary">
            {governorate + ', ' + location + ', ' + street}
          </Typography>
          <Link to={`/explore/${id}`}>
            <Button sx={{ width: '100%', marginBottom: '1rem' }} variant="contained">
              More Details
            </Button>
          </Link>
        </Stack>
      </Paper>
    </>
  )
}

export default Warehouse
