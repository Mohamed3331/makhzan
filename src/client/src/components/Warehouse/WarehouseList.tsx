import React from 'react'
import { Skeleton, Stack } from '@mui/material'

import { Box } from '@mui/system'

import '../../styles/WarehouseList.css'
import 'swiper/css'
import { WarehouseAttributes } from '../../types'
import Warehouse from './Warehouse'

export interface IWarehousesListState {
  warehouses: object[]
  loading: boolean
  error: string
}

const WarehouseList: React.FC<IWarehousesListState> = ({ warehouses = [], loading, error }) => {
  if (loading) {
    return (
      <>
        {warehouses.map(() => {
          return (
            <>
              <Stack spacing={1}>
                <Box sx={{ display: 'flex', justifyContent: 'space-around', alignItems: 'center' }}>
                  <Skeleton variant="circular" width={150} height={150} />
                  <Skeleton variant="rectangular" width={600} height={140} />
                </Box>
              </Stack>
            </>
          )
        })}
      </>
    )
  }

  if (warehouses.length === 0) {
    return <div>No warehouses to Display</div>
  }

  if (error) {
    return <div>{error}</div>
  }

  return (
    <>
      <Stack spacing={5}>
        {warehouses.map((warehouse: any) => {
          return (
            <div key={warehouse?.id}>
              <Box sx={{ maxWidth: '100%' }}>
                <Warehouse {...warehouse} />
              </Box>
            </div>
          )
        })}
      </Stack>
    </>
  )
}

export default WarehouseList
