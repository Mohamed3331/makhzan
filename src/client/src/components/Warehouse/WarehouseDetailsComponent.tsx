import { Box, Container, ImageListItem, Grid } from '@mui/material'
import React from 'react'
import BreadCrumbsLinks from '../BreadCrumbsLinks'
import Image1 from '../../Assets/image1.jpeg'
import Image2 from '../../Assets/image2.jpeg'
import Image3 from '../../Assets/image3.jpeg'
import LandingNavbar from '../Navbar/NavbarComponent'
import WarehouseDescription from './WarehouseDescription'

const WarehouseDetailsComponent: React.FC = () => {
  return (
    <>
      <LandingNavbar Position="relative" />
      <Container>
        <BreadCrumbsLinks text="fds" link="/" />
        <Box sx={{ height: '100vh' }}>
          <Grid columnSpacing={{ xs: 1, sm: 2, md: 1 }} container>
            <Grid item xs={8}>
              <ImageListItem>
                <img src={Image1} srcSet={Image1} alt={'fdsdsfds'} loading="lazy" />
              </ImageListItem>
            </Grid>
            <Grid item xs={4}>
              <ImageListItem>
                <img src={Image2} srcSet={Image2} alt={'fdsdsfds'} loading="lazy" />
              </ImageListItem>
              <ImageListItem>
                <img src={Image3} srcSet={Image3} alt={'fdsdsfds'} loading="lazy" />
              </ImageListItem>
            </Grid>
          </Grid>
          <WarehouseDescription />
        </Box>
        v
      </Container>
    </>
  )
}

export default WarehouseDetailsComponent
