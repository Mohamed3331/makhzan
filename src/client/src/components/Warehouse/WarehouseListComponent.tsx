import React, { useState } from 'react'
import { Typography, Paper, Button, Skeleton, Stack } from '@mui/material'
import { Box } from '@mui/system'

import WarehouseComponent from './WarehouseComponent'
import 'swiper/css'
import '../../styles/WarehouseList.css'

const WarehouseList = () => {
  const [loading, setLoading] = useState(true)

  return (
    <>
      <WarehouseComponent />
    </>
  )
}

export default WarehouseList

//  <Stack spacing={4}>
// <Box sx={{ display: 'flex', justifyContent: 'center' }}>
//   <Card sx={{ display: 'flex', direction: 'row', borderRadius: '1px', maxWidth: '80rem' }}>
//     <img height={SwiperImageHeight} src={Image1} alt="" />
// {/* <Swiper
//   className="swiper-override-style"
//   spaceBetween={0}
//   slidesPerView={1}
//   onSlideChange={() => console.log('slide change')}
//   onSwiper={(swiper) => console.log(swiper)}
//   pagination={{ clickable: true }}
//   loop
// >
//   <SwiperSlide>
//     <img height={SwiperImageHeight} src={Image1} alt="" />
//   </SwiperSlide>
//   <SwiperSlide>
//     <img height={SwiperImageHeight} src={Image2} alt="" />
//   </SwiperSlide>
//   <SwiperSlide>
//     <img height={SwiperImageHeight} src={Image3} alt="" />
//   </SwiperSlide>
// </Swiper> */}
//     <Box sx={{ display: 'flex', flexDirection: 'column' }}>
//       <CardContent>
// <Typography gutterBottom variant="h5" component="div">
//   3,150,000
// </Typography>
// <Typography variant="body2" color="text.secondary">
//   hadayek el ahraam, street 105, apartment 15
// </Typography>
//       </CardContent>

//       <Button size="small">Share</Button>
//       <Button size="small">Learn More</Button>
//     </Box>
//   </Card>
// </Box>
// </Stack>
