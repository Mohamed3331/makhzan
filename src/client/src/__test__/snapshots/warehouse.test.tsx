import { MemoryRouter } from 'react-router-dom'
import renderer from 'react-test-renderer'
import Warehouse from '../../components/WarehousesList/Warehouse'
import { warehouses } from '../TestHelpers/fixtures'

test('warehouse component snapshot renders correctly', () => {
  const tree = renderer.create(
    <MemoryRouter>
      <Warehouse {...warehouses[0]} />
    </MemoryRouter>
  )
  expect(tree).toMatchSnapshot()
})
