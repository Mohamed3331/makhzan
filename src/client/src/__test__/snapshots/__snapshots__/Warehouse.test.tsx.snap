// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`warehouse component snapshot renders correctly 1`] = `
<div
  className="MuiPaper-root MuiPaper-outlined css-16qmdhu-MuiPaper-root"
  data-testid="warehouse-node"
>
  <div
    className="warehouse-img-wrapper MuiBox-root css-0"
  >
    <img
      alt=""
      src="https://res.cloudinary.com/makhzan/image/upload/v1643349711/cld-sample.jpg"
      style={
        Object {
          "maxWidth": "27rem",
        }
      }
    />
  </div>
  <div
    className="css-1arhqv8-MuiStack-root"
  >
    <h5
      className="MuiTypography-root MuiTypography-h5 css-ag7rrr-MuiTypography-root"
    >
      150,000
       EGP / Month
    </h5>
    <p
      className="MuiTypography-root MuiTypography-body2 css-e784if-MuiTypography-root"
    >
      Warehouse 1500 SQM licensed Food & Beverage
    </p>
    <div
      className="css-e53awj-MuiStack-root"
    >
      <div
        className="MuiBox-root css-70qvj9"
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
          data-testid="WarehouseOutlinedIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M20 8.35V19h-2v-8H6v8H4V8.35l8-3.2 8 3.2zM22 21V7L12 3 2 7v14h6v-8h8v8h6zm-11-2H9v2h2v-2zm2-3h-2v2h2v-2zm2 3h-2v2h2v-2z"
          />
        </svg>
        <p
          className="MuiTypography-root MuiTypography-body2 css-e784if-MuiTypography-root"
        >
          Warehouse
        </p>
      </div>
      <div
        className="MuiBox-root css-70qvj9"
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
          data-testid="SquareFootSharpIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="m17.66 17.66-1.06 1.06-.71-.71 1.06-1.06-1.94-1.94-1.06 1.06-.71-.71 1.06-1.06-1.94-1.94-1.06 1.06-.71-.71 1.06-1.06L9.7 9.7l-1.06 1.06-.71-.71 1.06-1.06-1.94-1.94-1.06 1.06-.71-.71 1.06-1.06L4 4v16h16l-2.34-2.34zM7 17v-5.76L12.76 17H7z"
          />
        </svg>
        <p
          className="MuiTypography-root MuiTypography-body2 css-e784if-MuiTypography-root"
        >
          1500
           sqm
        </p>
      </div>
    </div>
    <p
      className="MuiTypography-root MuiTypography-body2 css-e784if-MuiTypography-root"
    >
      Cairo , 6 October City , 112
    </p>
    <a
      href="/explore/202a085b-4c55-4d7e-9cc1-f2169730256d"
      onClick={[Function]}
    >
      <button
        className="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButtonBase-root  css-1umhlf7-MuiButtonBase-root-MuiButton-root"
        disabled={false}
        onBlur={[Function]}
        onContextMenu={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        More Details
        <span
          className="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
    </a>
  </div>
</div>
`;
